net/mwan3/files/usr/sbin/mwan3

```diff
@@ -81,8 +81,16 @@ ifup()
 	if [ "$up" = "1" ] \
 		&& [ -n "$l3_device" ] \
 		&& [ "$enabled" = "1" ]; then
-		ACTION=ifup INTERFACE=$1 DEVICE=$l3_device /sbin/hotplug-call iface
```

Comment:
Sorry this does not work! I want to have this.
All hotplug script must be called under /etc/hotplug.d/iface I have additional scripts there that should also get notified.

Reply:
We could move MWAN3_STARTUP into **/var/run/mwan3/startup** ? so we do not needed to change the hotplug.d call

Reply:
@feckert Can you describe your setup a bit more? My problem was that I had a lot of stuff in the hotplug directory that shouldn't have been called on an `mwan3 start`.  

Connecting an interface for the first time requires a different response than initiating mwan3. So it didn't really make sense to me to cause a full ifup hotplug event when a user runs '/init.d/mwan3/restart.'  For example, I have a hotplug script that brings up my VPN when the router connects to the 4G modem and gets an ifup event. With the current configuration, it will try to restart the VPN when mwan3 is restarted. This slowed down the mwan3 startup process and seemed unnecessary.

>We could also add this check to the ubus backend.

I don't understand, could you explain a bit more?

> We could move MWAN3_STARTUP into /var/run/mwan3/startup ?

Could you explain a bit more? It would be nice to avoid the MWAN3_STARTUP semaphore if possible.

Reply:
> @feckert Can you describe your setup a bit more? My problem was that I had a lot of stuff in the hotplug directory that shouldn't have been called on an `mwan3 start`.

Okay, now I get it. I missed you explaining why you wanted to do it that way.

> Connecting an interface for the first time requires a different response than initiating mwan3. So it didn't really make sense to me to cause a full ifup hotplug event when a user runs '/init.d/mwan3/restart.' For example, I have a hotplug script that brings up my VPN when the router connects to the 4G modem and gets an ifup event. With the current configuration, it will try to restart the VPN when mwan3 is restarted. This slowed down the mwan3 startup process and seemed unnecessary.

Sounds reasonable. Fits from my side.